operators/src/pro/cache/cache_operator.rs

```diff
@@ -359,33 +358,32 @@ trait ResultStreamWrapper: CacheElement {
 
 impl<G> ResultStreamWrapper for FeatureCollection<G>
 where
-    G: Geometry + ArrowTyped + Send + Sync + 'static,
-    FeatureCollection<G>: CacheElement,
-    <FeatureCollection<G> as CacheElement>::ResultStream: FusedStream + Send,
-    <FeatureCollection<G> as CacheElement>::ResultStream: Stream<Item = Result<Self>>,
+    G: Geometry + ArrowTyped + Send + Sync + 'static + CacheElementSubType,
+    FeatureCollection<G>: CacheElement + Send + Sync,
+    Self::ResultStream: FusedStream + Send + Sync,
 {
     fn wrap_result_stream<'a>(
         stream: Self::ResultStream,
         chunk_byte_size: ChunkByteSize,
     ) -> BoxStream<'a, Result<Self>> {
         Box::pin(FeatureCollectionChunkMerger::new(
-            stream,
+            stream.map_err(|ce| Error::CacheCantProduceResult { source: ce.into() }),
```

Comment:
how can there be errors in the first place?

Reply:
i don't unterstand your question. The cache produces a stream and there might be problems when producing elements e.g. when decompressing them.